flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/StreamTwoInputProcessorFactory.java

```diff
@@ -40,38 +48,78 @@
  */
 public class StreamTwoInputProcessorFactory {
 	public static <IN1, IN2> StreamTwoInputProcessor<IN1, IN2> create(
+			AbstractInvokable ownerTask,
 			CheckpointedInputGate[] checkpointedInputGates,
-			TypeSerializer<IN1> inputSerializer1,
-			TypeSerializer<IN2> inputSerializer2,
 			IOManager ioManager,
+			MemoryManager memoryManager,
 			TaskIOMetricGroup taskIOMetricGroup,
 			StreamStatusMaintainer streamStatusMaintainer,
 			TwoInputStreamOperator<IN1, IN2, ?> streamOperator,
 			TwoInputSelectionHandler inputSelectionHandler,
 			WatermarkGauge input1WatermarkGauge,
 			WatermarkGauge input2WatermarkGauge,
 			OperatorChain<?, ?> operatorChain,
+			StreamConfig streamConfig,
+			Configuration taskManagerConfig,
+			Configuration jobConfig,
+			ExecutionConfig executionConfig,
+			ClassLoader userClassloader,
 			Counter numRecordsIn) {
 
 		checkNotNull(operatorChain);
 		checkNotNull(inputSelectionHandler);
+
 		StreamStatusTracker statusTracker = new StreamStatusTracker();
 		taskIOMetricGroup.reuseRecordsInputCounter(numRecordsIn);
+		TypeSerializer<IN1> typeSerializer1 = streamConfig.getTypeSerializerIn(0, userClassloader);
+		StreamTaskInput<IN1> input1 = new StreamTaskNetworkInput<>(
+			checkpointedInputGates[0],
+			typeSerializer1,
+			ioManager,
+			new StatusWatermarkValve(checkpointedInputGates[0].getNumberOfInputChannels()),
+			0);
+		TypeSerializer<IN2> typeSerializer2 = streamConfig.getTypeSerializerIn(1, userClassloader);
+		StreamTaskInput<IN2> input2 = new StreamTaskNetworkInput<>(
+			checkpointedInputGates[1],
+			typeSerializer2,
+			ioManager,
+			new StatusWatermarkValve(checkpointedInputGates[1].getNumberOfInputChannels()),
+			1);
+
+		if (streamConfig.shouldSortInputs()) {
+			@SuppressWarnings("unchecked")
+			MultiInputSortingDataInputs<Object> multiInputs = new MultiInputSortingDataInputs<Object>(
```

Comment:
(I remember someone complaining about unchecked warnings 🙈 and `Object`? 😈  )

Reply:
Yes, I must admit I was a bit harsh on this one :sweat_smile: